$NetBSD: patch-aa,v 1.11 2017/01/25 15:53:06 adam Exp $

Fix ifdef/endif.
Use LIBTOOL.
Remove duplicate objects (one line).

--- SRC/Makefile.orig	2016-12-23 23:01:32.000000000 +0000
+++ SRC/Makefile
@@ -174,7 +174,7 @@ SLASRC = \
 
 DSLASRC = spotrs.o sgetrs.o spotrf.o sgetrf.o
 
-ifdef USEXBLAS
+.ifdef USEXBLAS
 SXLASRC = sgesvxx.o sgerfsx.o sla_gerfsx_extended.o sla_geamv.o \
    sla_gercond.o sla_gerpvgrw.o ssysvxx.o ssyrfsx.o \
    sla_syrfsx_extended.o sla_syamv.o sla_syrcond.o sla_syrpvgrw.o \
@@ -182,7 +182,7 @@ SXLASRC = sgesvxx.o sgerfsx.o sla_gerfsx
    sla_porpvgrw.o sgbsvxx.o sgbrfsx.o sla_gbrfsx_extended.o \
    sla_gbamv.o sla_gbrcond.o sla_gbrpvgrw.o sla_lin_berr.o slarscl2.o \
    slascl2.o sla_wwaddw.o
-endif
+.endif
 
 CLASRC = \
    cpotrf2.o cgetrf2.o \
@@ -271,7 +271,7 @@ CLASRC = \
    cheevd_2stage.o cheev_2stage.o cheevx_2stage.o cheevr_2stage.o \
    chbev_2stage.o chbevx_2stage.o chbevd_2stage.o chegv_2stage.o
 
-ifdef USEXBLAS
+.ifdef USEXBLAS
 CXLASRC = cgesvxx.o cgerfsx.o cla_gerfsx_extended.o cla_geamv.o \
    cla_gercond_c.o cla_gercond_x.o cla_gerpvgrw.o \
    csysvxx.o csyrfsx.o cla_syrfsx_extended.o cla_syamv.o \
@@ -283,7 +283,7 @@ CXLASRC = cgesvxx.o cgerfsx.o cla_gerfsx
    chesvxx.o cherfsx.o cla_herfsx_extended.o cla_heamv.o \
    cla_hercond_c.o cla_hercond_x.o cla_herpvgrw.o \
    cla_lin_berr.o clarscl2.o clascl2.o cla_wwaddw.o
-endif
+.endif
 
 ZCLASRC = cpotrs.o cgetrs.o cpotrf.o cgetrf.o
 
@@ -366,7 +366,7 @@ DLASRC = \
    dsyevd_2stage.o dsyev_2stage.o dsyevx_2stage.o dsyevr_2stage.o \
    dsbev_2stage.o dsbevx_2stage.o dsbevd_2stage.o dsygv_2stage.o
 
-ifdef USEXBLAS
+.ifdef USEXBLAS
 DXLASRC = dgesvxx.o dgerfsx.o dla_gerfsx_extended.o dla_geamv.o \
    dla_gercond.o dla_gerpvgrw.o dsysvxx.o dsyrfsx.o \
    dla_syrfsx_extended.o dla_syamv.o dla_syrcond.o dla_syrpvgrw.o \
@@ -374,7 +374,7 @@ DXLASRC = dgesvxx.o dgerfsx.o dla_gerfsx
    dla_porpvgrw.o dgbsvxx.o dgbrfsx.o dla_gbrfsx_extended.o \
    dla_gbamv.o dla_gbrcond.o dla_gbrpvgrw.o dla_lin_berr.o dlarscl2.o \
    dlascl2.o dla_wwaddw.o
-endif
+.endif
 
 ZLASRC = \
    zpotrf2.o zgetrf2.o \
@@ -464,12 +464,11 @@ ZLASRC = \
    zgelqt.o zgelqt3.o zgemlqt.o \
    zgetsls.o zgeqr.o zlatsqr.o zlamtsqr.o zgemqr.o \
    zgelq.o zlaswlq.o zlamswlq.o zgemlq.o \
-   ztplqt.o ztplqt2.o ztpmlqt.o \
    zhetrd_2stage.o zhetrd_he2hb.o zhetrd_hb2st.o zhb2st_kernels.o \
    zheevd_2stage.o zheev_2stage.o zheevx_2stage.o zheevr_2stage.o \
    zhbev_2stage.o zhbevx_2stage.o zhbevd_2stage.o zhegv_2stage.o
 
-ifdef USEXBLAS
+.ifdef USEXBLAS
 ZXLASRC = zgesvxx.o zgerfsx.o zla_gerfsx_extended.o zla_geamv.o \
    zla_gercond_c.o zla_gercond_x.o zla_gerpvgrw.o zsysvxx.o zsyrfsx.o \
    zla_syrfsx_extended.o zla_syamv.o zla_syrcond_c.o zla_syrcond_x.o \
@@ -479,7 +478,7 @@ ZXLASRC = zgesvxx.o zgerfsx.o zla_gerfsx
    zla_gbrpvgrw.o zhesvxx.o zherfsx.o zla_herfsx_extended.o \
    zla_heamv.o zla_hercond_c.o zla_hercond_x.o zla_herpvgrw.o \
    zla_lin_berr.o zlarscl2.o zlascl2.o zla_wwaddw.o
-endif
+.endif
 
 DEPRECSRC = DEPRECATED/cgegs.o DEPRECATED/cgegv.o DEPRECATED/cgelsx.o \
    DEPRECATED/cgeqpf.o DEPRECATED/cggsvd.o DEPRECATED/cggsvp.o \
@@ -497,19 +496,19 @@ DEPRECSRC = DEPRECATED/cgegs.o DEPRECATE
 ALLOBJ = $(SLASRC) $(DLASRC) $(DSLASRC) $(CLASRC) $(ZLASRC) $(ZCLASRC) \
    $(SCLAUX) $(DZLAUX) $(ALLAUX)
 
-ifdef USEXBLAS
+.ifdef USEXBLAS
 ALLXOBJ = $(SXLASRC) $(DXLASRC) $(CXLASRC) $(ZXLASRC)
-endif
+.endif
 
-ifdef BUILD_DEPRECATED
+.ifdef BUILD_DEPRECATED
 DEPRECATED = $(DEPRECSRC)
-endif
+.endif
 
 all: ../$(LAPACKLIB)
 
 ../$(LAPACKLIB): $(ALLOBJ) $(ALLXOBJ) $(DEPRECATED)
-	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ) $(ALLXOBJ) $(DEPRECATED)
-	$(RANLIB) $@
+	${LIBTOOL} --mode=link --tag=FC ${FC} ${LDFLAGS} -o ../$(LAPACKLIB) \
+	$(ALLOBJ:.o=.lo) -rpath ${PREFIX}/lib -version-info 4:0
 
 single: $(SLASRC) $(DSLASRC) $(SXLASRC) $(SCLAUX) $(ALLAUX)
 	$(ARCH) $(ARCHFLAGS) ../$(LAPACKLIB) $(SLASRC) $(DSLASRC) \
@@ -540,12 +539,12 @@ $(DLASRC): $(FRC)
 $(ZLASRC): $(FRC)
 $(ZCLASRC): $(FRC)
 $(DSLASRC): $(FRC)
-ifdef USEXBLAS
+.ifdef USEXBLAS
 $(SXLASRC): $(FRC)
 $(CXLASRC): $(FRC)
 $(DXLASRC): $(FRC)
 $(ZXLASRC): $(FRC)
-endif
+.endif
 
 FRC:
 	@FRC=$(FRC)
@@ -554,14 +553,14 @@ clean:
 	rm -f *.o DEPRECATED/*.o
 
 .f.o:
-	$(FORTRAN) $(OPTS) -c -o $@ $<
+	${LIBTOOL} --mode=compile --tag=FC ${FC} ${FFLAGS} -c $<
 
 .F.o:
-	$(FORTRAN) $(OPTS) -c $< -o $@
+	${LIBTOOL} --mode=compile --tag=FC ${FC} ${FFLAGS} -c $<
 	
-slaruv.o: slaruv.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
-dlaruv.o: dlaruv.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
-sla_wwaddw.o: sla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
-dla_wwaddw.o: dla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
-cla_wwaddw.o: cla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
-zla_wwaddw.o: zla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
+slaruv.o: slaruv.f ; ${LIBTOOL} --mode=compile --tag=FC $(FC) $(NOOPT) -c $< -o $@
+dlaruv.o: dlaruv.f ; ${LIBTOOL} --mode=compile --tag=FC $(FC) $(NOOPT) -c $< -o $@
+sla_wwaddw.o: sla_wwaddw.f ; ${LIBTOOL} --mode=compile --tag=FC $(FC) $(NOOPT) -c $< -o $@
+dla_wwaddw.o: dla_wwaddw.f ; ${LIBTOOL} --mode=compile --tag=FC $(FC) $(NOOPT) -c $< -o $@
+cla_wwaddw.o: cla_wwaddw.f ; ${LIBTOOL} --mode=compile --tag=FC $(FC) $(NOOPT) -c $< -o $@
+zla_wwaddw.o: zla_wwaddw.f ; ${LIBTOOL} --mode=compile --tag=FC $(FC) $(NOOPT) -c $< -o $@
